### ISSUE TITLE
Feature request: Connect a partner without having children

### ISSUE THREAD

**Issue by cloetensbrecht** [BasicPrimitives/react#72]
It would be great to have the possibility to add a partner to the FamDiagram. It seems only possible to connect items through their parents. But if you want to connect an item as a partner it seems not possible. You can use the annotations to draw a line between the partners, but within a family tree it's more obvious to add partners next to eachother. 

Ex: connect them with an empty child:
![image](https://user-images.githubusercontent.com/1882256/217025887-54eddd45-ff09-4f24-8c9e-80a79adf9732.png)

Desired connection:
![image](https://user-images.githubusercontent.com/1882256/217027833-098d3675-9028-440f-8b13-6d8ba4fd83cc.png)

**Comment by BasicPrimitives** [comment]
The "horizontal connection feature" is only applicable for family diagrams centered around a single person's ancestry. Our diagramming component is called "Family Diagram" because it is the closest to a layered dependency graph visualization. In cases where multiple families are merged into one large family or in complex legal cases with numerous relatives, the regular diagram may not be effective. As a result, we have decided not to include horizontal connection lines. 

For example, in a basic 3-parents family consisting of a mother, father, and mitochondrial DNA donor, they may have children in multiple inheritance lines and therefore cannot be placed side by side to display a horizontal connection line between them. The same applies when the mother or father have children with multiple spouses; the classic family diagram is designed only for displaying single person ancestors. As a result, we do not support or plan to support horizontal connection lines between partners.

**Comment by TheGuayeah** [comment]
Hi BasicPrimitives Team,
I was also expecting to connect partners without children, but in my case instead of horizontal lines it could be done like in the following example:
![image](https://github.com/BasicPrimitives/react/assets/37774329/e2ceb90a-c13b-47d6-8b33-5acbe0d8cb11)
This behaviour is highly desirable for my family tree as it looks more clear that those relatives have a visual connection on this diagram.

In the other hand, the horizontal line connection is also good to represent other information such as if two or more siblings are twins. This can let them stand out from regular siblings

**Comment by miri-cmkl** [comment]
We would also very much like to request this feature: _Ability to connect a partner without having children_

Currently, this library enables us to create family trees by displaying how each relative is inter-connected. However, these connections are not visually complete...

When we create a new sibling or parent, we can clearly see in the tree that each of them has been positioned and connected through a line in the graph, though we don't see the same behavior while adding a spouse/partner.

Creating a spouse doesn't place the newly created relative in the corresponding position, nor does it create a line to connect them unless there is already a child created with the corresponding "relative Id" in the JSON file.

This is not desirable in some cases as, for example, we would like to be able to show at all time who is the spouse of whom, regardless of if they have a child in common or not.

In other words, in some instances, this is undesirable; for instance, it's preferable to consistently display spousal relationships, irrespective of whether they share a child.

It would be great, and it is our humble request, to have an option that additionally connects two or more siblings if they happen to be twins.

This is quite similar to the desired connection described and requested by [cloetensbrecht](https://github.com/cloetensbrecht)
